src/client.coffee

```diff
@@ -169,28 +174,32 @@ class SlackClient
       fetches = {};
       fetches.user = @web.users.info(event.user) if event.user
       fetches.channel = @web.conversations.info(event.channel) if event.channel
-      fetches.bot = @web.bots.info(event.bot_id) if event.bot_id
       fetches.item_user = @web.users.info(event.item_user) if event.item_user
 
       Promise.props(fetches).then((fetched) =>
 
         event.channel = fetched.channel if fetched.channel
 
-        # messages sent from human users, apps with a bot user and using the xoxb token, and
-        # slackbot have the user property
-        event.user = fetched.user if fetched.user
-
-        # bot_id exists on all messages with subtype bot_message
-        # these messages only have a user property if sent from a bot user (xoxb token). therefore
-        # the above assignment will not happen for all messages from custom integrations or apps without a bot user
-        event.bot = fetched.bot if fetched.bot
-
         # this property is for reaction_added and reaction_removed events
         # previous behavior was to ignore this event entirely if the user and item_user were not in the local workspace
         event.item_user = fetched.item_user if fetched.item_user
 
-        try @eventHandler(event)
-        catch error then @robot.logger.error "An error occurred while processing an RTM event: #{error.message}."
+        # messages sent from human users, apps with a bot user and using the xoxb token, and
+        # slackbot have the user property
+        if fetched.user
+          event.user = fetched.user
+          try @eventHandler(event)
+          catch error then @robot.logger.error "An error occurred while processing an RTM event: #{error.message}."
+        # User always preferred over bot
+        else if event.bot_id
+          # bot_id exists on all messages with subtype bot_message
+          # these messages only have a user property if sent from a bot user (xoxb token). therefore
+          # the above assignment will not happen for all messages from custom integrations or apps without a bot user
+          @loadUsers(event.bot_id, (err, res) =>
+            event.user = res
```

Comment:
what happens if `res` is `undefined`? that's a possibility according to how `loadUsers` is implemented.